### PR TITLE
package.json: upgrade patternfly to version 5

### PR DESCRIPTION
**Pull request by jkozol**
This upgrade requires multiple package updates. There are many style updates to reflect pf5's syntax and structure changes.

### CODE REVIEW

File: src/Components/ShareImageModal/RegionsSelect.tsx
Comment (lucasgarfield) [inline src/Components/ShareImageModal/RegionsSelect.tsx:103]: We discussed this offline, just adding a reminder here so I remember to double check myself that this was removed (multiple select shouldn't close open selecting an option).

File: src/Components/ShareImageModal/RegionsSelect.tsx
Comment (lucasgarfield) [inline src/Components/ShareImageModal/RegionsSelect.tsx:107]: Reminder to remove.

**Comment by jrusz** [comment]
I found an issue with the "Select region" `Select`. When I click on the button to open the item list I can't close it again by clicking on that button, I have to click away to some empty space.

**Comment by jkozol** [comment]
@jrusz should be fixed now.

**Comment by lucasgarfield** [comment]
Looking good! I think the last thing that needs a bit of work is the validation on the share to regions modal:

this PR
image

vs

expected
image

**Comment by lavocatt** [comment]
The custom CSS rules to have the wizard in page behaving somewhat correctly are needing an update:

You to change the last bits of the `CreateImageWizard.scss` file with:

```
// [2023-09] Wizard height should not be limited see
// https://github.com/data-driven-forms/react-forms/issues/1402
// TODO remove me after DDF is not used anymore
.pf-v5-c-wizard__main {
    overflow-x: unset;
    overflow-y: unset;
}
// TODO remove me after DDF is not used anymore
.pf-v5-c-wizard__footer {
    z-index: unset;
}
// [2023-10] Wizard closing cross shouldn't get displayed
// TODO remove me after DDF is not used anymore
.pf-v5-c-wizard__header .pf-v5-c-button.pf-m-plain {
    display: none;
}
```

**Comment by lavocatt** [comment]
The navigation bar of the wizard now have a horizontal scroll bar:
image

**Comment by regexowl** [comment]
> The table is behaving strangely, like there is a line under /tmp and /home. It wasn't there with PF4

@lavocatt In the PF4 version the textboxes are next to the mount points and allow further customization. They're just out of place here, which should be hopefully solvable with some column width setting. :wink: 

image

**Comment by lavocatt** [comment]
> > The table is behaving strangely, like there is a line under /tmp and /home. It wasn't there with PF4
> 
> @lavocatt In the PF4 version the textboxes are next to the mount points and allow further customization. They're just out of place here, which should be hopefully solvable with some column width setting. 😉
> 
Ah! silly me! Thanks for the clarification! :smile:

**Comment by lavocatt** [comment]
Ahhh sorry I've got another annoying one :/

[Screencast from 2023-10-11 16-36-34.webm](https://github.com/RedHatInsights/image-builder-frontend/assets/86971992/2122d331-38b4-4160-b9b8-6b1d86c99ca9)

So what's happening is that when you create an image against multiple upload target, the navigation menu bugs after. So if you go through all the steps and then try to navigate your way in the navigation menu, you'll notice that where you click is not where you endup.

I've tried to get a video for this but my mouse pointer isn't showing up! So I hope the description still helps you reproducing this one.

**Comment by jkozol** [comment]
@lucasgarfield your issue should be fixed

@regexowl @lavocatt Can we move these to follow up PRs. Rebasing this is taking too much time.

Also, @lavocatt if you want to add changes please do so in a seperate commit.

**Comment by regexowl** [comment]
@jkozol Moving the "cosmetic" bugs to follow up PRs works perfectly fine for me!

**Comment by jrusz** [comment]
I was wondering. In the region select, the main item and description got swapped, is that intentional? See screenshots
image
image

**Comment by lucasgarfield** [comment]
I'm okay with issuing follow-up PRs but I don't think we should merge until we have the navigation bug Thomas found fixed. It's really bizarre, I'm very surprised that this PR would cause a bug like that but I can't reproduce the bug in main (and I can reproduce it for this PR).

**Comment by jrusz** [comment]
Also unit tests are being stubborn in pr_check again...

**Comment by lucasgarfield** [comment]
I think I may have figured out what is causing the navigation bug, it seems like it is a DDF bug leftover from when they migrated to PF5, I issued a PR:
https://github.com/data-driven-forms/react-forms/pull/1414

**Comment by lucasgarfield** [comment]
#1388 bumps DDF to a new version which has the hotfix for the navigation issue. We should be good to go now.

**Comment by lucasgarfield** [comment]
Added blocked tag. Please do not merge until we have deployed. I want to get some minor adjustments to the in-page wizard merged, deploy with those in place, then merge this and restart our regular 2 week release cadence.

**Comment by jkozol** [comment]
All issues should be fixed now except there is one more. The nav list no longer shows the step numbers. @lucasgarfield did you end up figuring out how to render our wizard using a local version of ddf since this seems like another bug to fix there?

**Comment by lucasgarfield** [comment]
> All issues should be fixed now except there is one more. The nav list no longer shows the step numbers. @lucasgarfield did you end up figuring out how to render our wizard using a local version of ddf since this seems like another bug to fix there?

No, when I submitted my patch to them I tried to build locally but couldn't get it to work... I just submitted the PR and was pleasantly surprised when I saw that my patch actually worked :sweat_smile:

**Comment by jrusz** [comment]
Unit tests in pr_check are failing on this.

**Comment by lucasgarfield** [comment]
Unblocked, the in-page wizard is in and we can deploy from that SHA. This can be merged as soon as tests pass and it has approval from a reviewer.

**Comment by jkozol** [comment]
https://github.com/data-driven-forms/react-forms/pull/1416 should fix the step numbering

**Comment by jkozol** [comment]
@jrusz why is this a failure since it seems the tests passed?

**Comment by jrusz** [comment]
> @jrusz why is this a failure since it seems the tests passed?

Some issue with deployment to ephemeral, looks like pulp again... Retrying to see if it's resolved.

**Comment by lucasgarfield** [comment]
Looks good to me, thank you @jkozol !

**Review by lucasgarfield — APPROVED** [review]

**Pull request merged**
